packages/web3-eth/src/errors.ts

```diff
@@ -1,21 +1,159 @@
+		);
+	}
+}
+
+export class MissingChainOrHardforkError extends Web3Error {
+	public constructor(value: { chain: string | undefined; hardfork: string | undefined }) {
+		// TODO - Discuss what to pass as value
+		super(
+			'N/A',
+			`When specifying chain and hardfork, both values must be defined. Received "chain": ${
+				value.chain ?? 'undefined'
+			}, "hardfork": ${value.hardfork ?? 'undefined'}`,
+		);
+	}
+}
+
+export class MissingGasError extends Web3Error {
+	public constructor(value: {
+		gas: Numbers | undefined;
+		gasLimit: Numbers | undefined;
+		maxPriorityFeePerGas: Numbers | undefined;
+		maxFeePerGas: Numbers | undefined;
+	}) {
+		super(
+			`gas: ${value.gas ?? 'undefined'}, gasLimit: ${
+				value.gasLimit ?? 'undefined'
+			}, maxPriorityFeePerGas: ${value.maxPriorityFeePerGas ?? 'undefined'}, maxFeePerGas: ${
+				value.maxFeePerGas ?? 'undefined'
+			}`,
+			'"gas" is missing',
+		);
+	}
+}
+
+export class InvalidGasOrGasPrice extends Web3Error {
+	public constructor(value: { gas: Numbers | undefined; gasPrice: Numbers | undefined }) {
+		super(
+			`gas: ${value.gas ?? 'undefined'}, gasPrice: ${value.gasPrice ?? 'undefined'}`,
+			'Gas or gasPrice is lower than 0',
+		);
+	}
+}
+
+export class InvalidMaxPriorityFeePerGasOrMaxFeePerGas extends Web3Error {
+	public constructor(value: {
+		maxPriorityFeePerGas: Numbers | undefined;
+		maxFeePerGas: Numbers | undefined;
+	}) {
+		super(
+			`maxPriorityFeePerGas: ${value.maxPriorityFeePerGas ?? 'undefined'}, maxFeePerGas: ${
+				value.maxFeePerGas ?? 'undefined'
+			}`,
+			'maxPriorityFeePerGas or maxFeePerGas is lower than 0',
+		);
+	}
+}
+
+export class Eip1559GasPriceError extends Web3Error {
+	public constructor(value: unknown) {
+		super(value, "eip-1559 transactions don't support gasPrice");
+	}
+}
+
+export class UnsupportedFeeMarketError extends Web3Error {
+	public constructor(value: {
+		maxPriorityFeePerGas: Numbers | undefined;
+		maxFeePerGas: Numbers | undefined;
+	}) {
+		super(
+			`maxPriorityFeePerGas: ${value.maxPriorityFeePerGas ?? 'undefined'}, maxFeePerGas: ${
+				value.maxFeePerGas ?? 'undefined'
+			}`,
+			"pre-eip-1559 transaction don't support maxFeePerGas/maxPriorityFeePerGas",
+		);
+	}
+}
+
+export class InvalidTransactionObjectError extends Web3Error {
+	public constructor(value: unknown) {
+		super(value, 'invalid transaction object');
+	}
+}
+
+export class InvalidNonceOrChainIdError extends Web3Error {
+	public constructor(value: { nonce: Numbers | undefined; chainId: Numbers | undefined }) {
+		super(
+			`nonce: ${value.nonce ?? 'undefined'}, chainId: ${value.chainId ?? 'undefined'}`,
+			'Nonce or chainId is lower than 0',
+		);
+	}
+}
+
+export class UnableToPopulateNonceError extends Web3Error {
+	public constructor() {
+		// TODO - Discuss what to pass as value
+		super('N/A', 'unable to populate nonce, no from address available');
```

Comment:
`Web3Error` requires a string value to be passed to be included in the error message. For this error it doesn't make sense to pass a value in and I wasn't sure what to pass to `Web3Error`'s constructor